third_party/move/move-binary-format/src/file_format.rs

```diff
@@ -1340,501 +1341,1256 @@ pub struct CodeUnit {
+        or the transaction will be aborted with an arithmetic error.
+    "#]
+    #[semantics = r#"
+        stack >> (rhs: u8)
+        stack >> (lhs: int_ty)
+        if rhs >= num_bits_in(int_ty)
+            arithmetic error
+        else
+            stack << (lhs __shift_left__ rhs)
+    "#]
+    #[paranoid_post = r#"
+        ty_stack >> left_ty 
+        ty_stack >> right_ty
+        ty_stack << left_ty
+    "#]
     Shl,
-    /// Shift the (second top value) right (top value) bits and pushes the result on the stack.
-    ///
-    /// Stack transition:
-    ///
-    /// ```..., u64_value(1), u64_value(2) -> ..., u64_value```
+
+    #[group = "bitwise"]
+    #[description = r#"
+        Shift the (second top value) left (top value) bits and pushes the result on the stack.
+
+        The number of bits shifted must be less than the number of bits in the integer value being shifted, 
+        or the transaction will be aborted with an arithmetic error.
+    "#]
+    #[semantics = r#"
+        stack >> (rhs: u8)
+        stack >> (lhs: int_ty)
+        if rhs >= num_bits_in(int_ty)
+            arithmetic error
+        else
+            stack << (lhs __shift_right__ rhs)
+    "#]
+    #[paranoid_post = r#"
+        ty_stack >> left_ty 
+        ty_stack >> right_ty
+        ty_stack << left_ty
+    "#]
     Shr,
-    /// Create a vector by packing a statically known number of elements from the stack. Abort the
-    /// execution if there are not enough number of elements on the stack to pack from or they don't
-    /// have the same type identified by the SignatureIndex.
-    ///
-    /// Stack transition:
-    ///
-    /// ```..., e1, e2, ..., eN -> ..., vec[e1, e2, ..., eN]```
+
+    #[group = "vector"]
+    #[description = r#"
+        Create a vector by packing a statically known number of elements from the stack. 
+
+        Abort the execution if there are not enough number of elements on the stack 
+        to pack from or they do not have the same type identified by the `elem_ty_idx`.
+    "#]
+    #[static_operands = "[elem_ty_idx] [n]"]
+    #[semantics = r#"
+        stack >> elem_n
+        ..
+        stack >> elem_1
+        stack << vector[elem_1, .., elem_n]
+    "#]
+    #[paranoid_post = r#"
+        elem_ty << instantiate elem_ty 
+        for i in 1..=n: 
+            ty_stack >> ty  
+            assert ty == elem_ty 
+        ty_stack << vector<elem_ty>
+    "#]
+    #[gas_type_creation_tier_0 = "elem_ty"]
     VecPack(SignatureIndex, u64),
-    /// Return the length of the vector,
-    ///
-    /// Stack transition:
-    ///
-    /// ```..., vector_reference -> ..., u64_value```
+
+    #[group = "vector"]
+    #[description = "Get the length of a vector."]
+    #[static_operands = "[elem_ty_idx]"]
+    #[semantics = r#"
+        stack >> vec_ref
+        stack << (*vec_ref).len
+    "#]
+    #[paranoid_post = r#"
+        elem_ty << instantiate elem_ty 
+        ty_stack >> ty 
+        assert ty == &elem_ty 
```

Comment:
From the semantics description, it seems like this should be a reference to a vector type, not reference to an element of vector?

Reply:
Yes, I'm just using the notion `&elem_ty` to indicate that this is a reference type created by prefixing `elem_ty` with `&`. 

Might be a bit confusing if you think of it in terms rust, but within the context of paranoid mode we only have types and no values so I guess it should be fine. Later we can take another look at the pseudocode syntax and see if we can improve it.